tests/README.md

```diff
@@ -0,0 +1,323 @@
+│    Template   │         │     workers      │         ╟─────────────────╢         │ Save to artifacts │
+└───────────────┘         └──────────────────┘         ║    has_readme   ║         └─────────┬─────────┘
+                                                       ╟─────────────────╢                   │
+                                                       ║ has_valid_dats  ║                   │
+                                                       ╟─────────────────╢                   │
+                                                       ║   datalas_get   ║                   ▼
+                                                       ╟─────────────────╢                   │
+                                                       ║ files_integrity ║                   │
+                                                       ╚═════════════════╝                   │
+                                                                                             │
+                                                  ┌─────────────────◄────────────────────────┘
+                                                  │
+                                                  ▼
+                                                  │
+                                               ╔══╧══╗
+                                               ║ END ║
+                                               ╚═════╝
+```
+
+The workflow is composed of two jobs: build and test.
+
+**Build job:**
+
+- No parallelism.
+- Install the dependencies and save them to the workspace for subsequent jobs.
+
+Before test job:<br/>
+The test suite is split by historical timing to run on multiple worker. This aims at reducing the total execution time.
+
+**Test job:**
+
+- Parallelism on 2 workers.
+- Execute the dataset template test suite.
+
+After test job: (WIP)<br/>
+The test results are parsed and save to CircleCI artifacts.
+#TODO Send them to a monitoring GitHub Repository
+
+## Life of a Dataset Test
+
+This section describes the different steps a dataset test will perform to validate the proper functioning of datasets.
+
+### Test creation
+
+At the start of the test job, the `template.py` file is used to generate a test file for each dataset.
+Changes made to a dataset should not influence the behavior of other datasets while changes to the test suite potentially impacts every dataset.
+To solve this, the test suite will start by retrieving all files modified during a pull request.
+Then, it will only test on the minimal set of datasets.
+This strategy aims at saving computing resources as well as time.
+
+To select whether a dataset should be tested or not, we follow this heuristic:
+
+1. File modified is **not** part of a dataset:
+   - is part of a whitelist (see below), then **ignore**;
+   - otherwise, **full rerun** of the test suite.
+2. File is part of a dataset, then **partial rerun** test for this dataset.
```

Comment:
You should explain that when we're not in a PR then all datasets are tested.

Reply:
Brought the change. Let me know if I should add further emphasis on this point